packages/tools/src/eventDispatchers/shared/getActiveToolForKeyboardEvent.ts

```diff
@@ -37,22 +37,29 @@ export default function getActiveToolForKeyboardEvent(
 
   const toolGroupToolNames = Object.keys(toolGroup.toolOptions);
 
-  for (let j = 0; j < toolGroupToolNames.length; j++) {
-    const toolName = toolGroupToolNames[j];
-    const toolOptions = toolGroup.toolOptions[toolName];
-
-    // tool has binding that matches the mouse button, if mouseEvent is undefined
-    // it uses the primary button
-    const correctBinding =
-      toolOptions.bindings.length &&
-      toolOptions.bindings.some(
-        (binding) =>
-          binding.mouseButton === (mouseButton ?? MouseBindings.Primary) &&
-          binding.modifierKey === modifierKey
-      );
-
-    if (toolOptions.mode === Active && correctBinding) {
-      return toolGroup.getToolInstance(toolName);
+  for (let i = 0; i < toolGroupToolNames.length; i++) {
```

Comment:
Consider refactoring this into a method on ToolGroupManager that finds a tool having a given function return true, and have it take as parameters the tool name, the instance name and the options for that tool.  Then, just use that for the getActiveTool...  - could make it either as a finder or a getAll method.  Then, the actual check method becomes fairly easy to write.